expression/builtin_string_test.go

```diff
@@ -983,3 +983,27 @@ func (s *testEvaluatorSuite) TestLpad(c *C) {
 		}
 	}
 }
+
+func (s *testEvaluatorSuite) TestQuote(c *C) {
+	defer testleak.AfterTest(c)()
+
+	tbl := []struct {
+		arg interface{}
+		ret interface{}
+	}{
+		{"Don\\'t!", `'Don\'t!'`},
```

Comment:
@ariesdevil 
You can use `` instead of "".

Please add the following test cases:
mysql> select quote("Don't");
+----------------+
| quote("Don't") |
+----------------+
| 'Don\'t'       |
+----------------+

mysql> select quote("Don\\'t");
+------------------+
| quote("Don\\'t") |
+------------------+
| 'Don\\\'t'       |
+------------------+ 

mysql> select quote("Don\"");                                                   
+----------------+
| quote("Don\"") |
+----------------+
| 'Don"'         |
+----------------+
1 row in set (0.00 sec)

mysql> select quote('Don\\\"');
+------------------+
| quote('Don\\\"') |
+------------------+
| 'Don\\"'         |
+------------------+
1 row in set (0.00 sec)